src/ase/cw/model/Server.java

```diff
@@ -81,12 +91,27 @@ public void startOrderProcess() {
         } else {
             System.out.println(getName() + "already started");
         }
+
+    }
+
+    @Override
+    public void pauseOrderProcess() {
+        // this does not pause the status immediately but waits until current order is processed
+        serverStatus = status.PAUSED;
+    }
+
+    @Override
+    public void restartOrderProcess() {
+        synchronized (serverThread) {
+            serverThread.notify();
+        }
+        setStatus(status.FREE);
     }
 
     /**
      * Stops the internal serverThread.
      * The Server will finish a started order process before stopping.
-     * This function will block, until the server finished the a started ordrer.
+     * This function will block, until the server finished the a started order.
```

Comment:
In the stopOrderProcess I made a bug: 
First  set stopThread = true;
Then call serverThread.interrupt();

Can  you please fix this as well?